src/irr_ptr.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
```

Comment:
I think you should add a license header.

Reply:
^

Reply:
Done, in 913fe82